File: iosApp/iosApp/data/mappers/media/AudioMapper.swift
//
//  AudioMapper.swift
//  iosApp
//
//  Created by nVoxel on 04.05.2024.
//  Copyright © 2024 orgName. All rights reserved.
//

import Foundation
import TDLibKit
import TGDriveKit

class AudioMapper {
    
    private let fileMapper: FileMapper
    
    init(fileMapper: FileMapper) {
        self.fileMapper = fileMapper
    }
    
    func toResponse(audio: Audio) -> CommonAudio {
        return CommonAudio(
            duration: Int32(audio.duration),
            title: audio.title,
            performer: audio.performer,
            fileName: audio.fileName,
            mimeType: audio.mimeType,
            file: fileMapper.toResponse(file: audio.audio)
        )
    }
}

File: source/data/mappers/src/androidMain/kotlin/com/voxeldev/tgdrive/mappers/media/AudioMapper.kt
package com.voxeldev.tgdrive.mappers.media

import org.drinkless.td.libcore.telegram.TdApi.Audio
import com.voxeldev.tgdrive.main.api.media.Audio as CommonAudio

/**
 * @author nvoxel
 */
class AudioMapper(
    private val fileMapper: FileMapper,
) {

    fun toResponse(audio: Audio): CommonAudio =
        CommonAudio(
            duration = audio.duration,
            title = audio.title,
            performer = audio.performer,
            fileName = audio.fileName,
            mimeType = audio.mimeType,
            file = fileMapper.toResponse(file = audio.audio),
        )
}
